go/tools/internal/stdlib_tags/stdlib_tags.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"go/build/constraint"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// Used to update the list of tags affecting the standard library kept in
+// transitions.bzl.
+func main() {
+	if len(os.Args) < 2 {
+		println("usage: stdlib_tags <go SDK src directory>...")
+		os.Exit(1)
+	}
+
+	tags := make(map[string]struct{})
+	for _, dir := range os.Args[1:] {
+		err := filepath.WalkDir(dir, func(path string, d os.DirEntry, err error) error {
+			if d.IsDir() {
+				if d.Name() == "testdata" {
+					return filepath.SkipDir
+				}
+				return nil
+			}
+			if filepath.Ext(path) != ".go" {
+				return nil
+			}
+			if strings.HasSuffix(filepath.Base(path), "_test.go") {
+				return nil
+			}
+			return walkFile(path, tags)
+		})
+		if err != nil {
+			println(dir + ": " + err.Error())
+			os.Exit(1)
+		}
+	}
+
+	var filteredTags []string
```

Comment:
We can pre-allocate the capacity of this for performance.
```suggestion
	filteredTags := make([]string, 0, len(tags))
```

Reply:
Done